### PR TITLE
Grafana 8 with js | Yet to test completely  

### PR DESCRIPTION
**Pull request by athulmurali**
USE this branch only for emergency grafana8 request and do not merge it until the migration is complete 

- Fixed all api calls  other   than query response 
- No issues with golang code atm 

**Todo**
-  Send over the test build to anyone possible and test. 
- Collect issues  and do the ts migration in parallel to this 

**Things to remember** 

- The other plugin from OCI - grafana logs  cannot be run in parallel to this as it doesn't support grafana 8  unless it it run in a separate grafana installation or instance.

**Known issues** : 

- On creating a new panel and creating a variable, if a variable is unused, it gives metric request error.  (error occurs occasionally) 
-  Once, the variable is used and then changed to a value that is not set to the variable, the error goes away. 

Screenshots : 

image

image

image

Sample build of grafana-8 with js 
/Users/athmural/Go/src/oci-grafana-metrics/oci-metrics-datasource-g8.zip

### CODE REVIEW

File: datasource.go [hunk 1]
```diff
@@ -3,600 +3,515 @@
...
+
+						frame := data.NewFrame(q.query.RefID,
+								data.NewField("Time", nil, []time.Time{}),
+								data.NewField("Value", nil, []float64{}).SetConfig(&data.FieldConfig{
+										DisplayNameFromDS: name,
+								}),
+						)
+
+						for _, metric := range item.AggregatedDatapoints {
+								frame.AppendRow(metric.Timestamp.UnixNano()/1000000, *(metric.Value))
```
Comment (wbrowne) [inline datasource.go:483]: @athulmurali Could you try this instead?

```suggestion
								frame.AppendRow(metric.Timestamp.Time, *(metric.Value))
```

**Comment by athulmurali** [comment]
@wbrowne  The backend now sends the timestamps perfectly. Appreciate your help

**Comment by athulmurali** [comment]
Thanks @ryantxu  for your inputs and help in the other PR for the same.

**Comment by MaikiGirardi** [comment]
Nothing yet guys? It's been 3 months since this issue was open.

**Pull request merged**